receiver/podmanreceiver/receiver.go

```diff
@@ -57,11 +66,73 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) registerMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
+	scrp, err := scraperhelper.NewScraper(typeStr, r.scrape, scraperhelper.WithStart(r.start))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	r.metricsComponent, err = scraperhelper.NewScraperControllerReceiver(&r.config.ScraperControllerSettings, set, mc, scraperhelper.AddScraper(scrp))
+	return err
+}
+
+func (r *receiver) registerLogsConsumer(lc consumer.Logs) {
+	r.logsConsumer = lc
+}
+
+func (r *receiver) Start(ctx context.Context, host component.Host) error {
+	// Check for logs pipeline
+	if r.logsConsumer == nil {
+		r.set.Logger.Warn("Logs receiver is not set")
+	} else {
+		eventBackoff := backoff.NewExponentialBackOff()
+		eventBackoff.InitialInterval = 2 * time.Second
+		eventBackoff.MaxInterval = 10 * time.Minute
+		eventBackoff.Multiplier = 2
+		eventBackoff.MaxElapsedTime = 0
+		r.isLogsShutdown = false
+		go func() {
+			// Retry if any errors occur while getting the events.
+			errorWhileRetry := backoff.Retry(func() error {
+				err := r.handleEvents(ctx, eventBackoff)
+				return err
+			}, eventBackoff)
+			if errorWhileRetry != nil {
+				r.set.Logger.Error("Retry failed", zap.Error(errorWhileRetry))
+			}
+		}()
+	}
+	// Check for metrics pipeline
+	if r.metricsConsumer == nil {
+		r.set.Logger.Warn("Metrics receiver is not set")
```

Comment:
Is this really a cause for warning? Are users expected to always configure this receiver with both logs and metrics pipelines?

Reply:
No, users are not expected to configure both but we have added the log to let the users know that they have not configured either logs or metrics pipeline. We got the [feedback](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5989#discussion_r740124264) on the same earlier.

Reply:
Isn't `INFO` sufficient?

Reply:
In my opinion, we should add it as a debug log because users would want to know while troubleshooting whether the metrics receiver is added or not. If we add it as an info log, they would get this log every time even if they purposely didn't add the metrics receiver.
cc: @owais

Reply:
I think INFO would be good.

Reply:
Done

Reply:
> In my opinion, we should add it as a debug log because users would want to know while troubleshooting whether the metrics receiver is added or not

If this is a user configuration error, the config should be validated at `Validate`. After that, warnings and infos should be issued only for one-time events, like connection failures or state changes. In this case, debug indeed sounds more suitable, provided this is not required to exist.

Reply:
> If this is a user configuration error, the config should be validated at `Validate`.
 
We can only validate in the `Validate` function if the configuration parameters are on the receiver's level like `endpoint` or `collection_interval`. In this case, it is a pipeline/service level configuration, which is available only on runtime. For this reason, we added the validation where it is available.

> In this case, debug indeed sounds more suitable, provided this is not required to exist.

@dmitryax @owais If everyone shares the same opinion then we can go with `DEBUG`

Reply:
Debug sounds good to me. It's expected that user would want only metrics to be collected

Reply:
Changed the log level to `Debug`.